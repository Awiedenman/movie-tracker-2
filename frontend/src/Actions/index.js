export const fetchInitialMovies = movies => ({
  type: 'FETCH_INITIAL_MOVIES',
  movies
});

export const userLogin = user => ({
  type: 'USER_LOGIN',
  id: user.id,
  name: user.name,
  loggedIn: user.loggedIn
});

export const userSignUp = newUserData => ({
  type: 'USER_SIGN_UP',
  id: newUserData.id
});

export const userSignOut = () => ({
  type: 'USER_SIGN_OUT'
});

export const clearUserFavorites = () => ({
  type: 'CLEAR_USER_FAVORITES'
});

export const addFavorite = movie => ({
  type: 'ADD_FAVORITE',
  favorite: movie
});

export const removeFavorite = () => ({
  type: 'REMOVE_FAVORITE'
});

export const addExistingFavorites = existingUserFavorites => {
  return {
    type: 'ADD_EXISTING_FAVORITES',
    favorites : existingUserFavorites
  };
};