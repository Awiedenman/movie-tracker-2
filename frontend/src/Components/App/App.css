.grid-container {
  display: grid;
  grid-template-areas: "header header header header" "main-container main-container main-container main-container" "main-container main-container main-container main-container";
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-template-rows: 9rem 1fr 1fr;
  height: 100%;
}

.header {
  align-items: center;
  border: 1px solid red;
  display: flex;
  grid-area: header;
  justify-content: space-between;
}

.main-container {
  border: 1px solid blue;
  grid-area: main-container;
}