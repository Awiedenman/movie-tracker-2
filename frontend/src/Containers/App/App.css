.grid-container {
  display: grid;
  grid-template-areas: "header header header header" "movie-list movie-list movie-list movie-list" "movie-list movie-list movie-list movie-list";
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-template-rows: 9rem 1fr 1fr;
  height: 100%;
}

.header {
  align-items: center;
  border: 1px solid red;
  display: flex;
  grid-area: header;
  justify-content: space-between;
}

.movie-list {
  border: 1px solid blue;
  grid-area: movie-list;
}