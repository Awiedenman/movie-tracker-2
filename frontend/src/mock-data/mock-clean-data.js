export const mockCleanedMovieList =[
  {
    average: 7.5,
    id: 284053,
    image: "/rzRwTcFvttcN1ZpX2xv4j3tSdJu.jpg",
    overview: "Thor is on the other side of the universe and finds himself in a race against time to get back to Asgard to stop Ragnarok, the prophecy of destruction to his homeworld and the end of Asgardian civilization, at the hands of an all-powerful new threat, the ruthless Hela.",
    title: "Thor: Ragnarok"
  }
];