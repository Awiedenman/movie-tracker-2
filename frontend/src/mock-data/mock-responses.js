export const mockInitialMovieResponse = {
  created_by: 'travisbell',
  description:'The idea behind this list is to collect the live action comic book movies from within the Marvel franchise. Last updated on Dec 18, 2013.',
  favorite_count: 0,
  id: '1',
  items: [
    {
      vote_average: 7.5,
      vote_count: 7168,
      id: 284053,
      video: false,
      media_type: 'movie',
      title: 'Thor: Ragnarok',
      popularity: 156.401,
      poster_path: '/rzRwTcFvttcN1ZpX2xv4j3tSdJu.jpg',
      original_language: 'en',
      original_title: 'Thor: Ragnarok',
      genre_ids: [28, 12, 14, 878, 35],
      backdrop_path: '/kaIfm5ryEOwYg8mLbq8HkPuM1Fo.jpg',
      adult: false,
      overview:'Thor is on the other side of the universe and finds himself in a race against time to get back to Asgard to stop Ragnarok, the prophecy of destruction to his homeworld and the end of Asgardian civilization, at the hands of an all-powerful new threat, the ruthless Hela.',
      release_date: '2017-10-25'
    }
  ]
};