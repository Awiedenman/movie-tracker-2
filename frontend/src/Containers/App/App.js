import React, { Component } from 'react';
import { connect } from 'react-redux';
import { movieFetch } from '../../helpers/apiCalls';
import { cleanMovieResponse } from '../../helpers/clean-responses';
import { fetchInitialMovies } from '../../Actions';
import PropTypes from 'prop-types';

export class App extends Component {
  constructor(props){
    super(props);
  }

  async componentDidMount(){
    const initialFetch = await movieFetch();
    const movies = cleanMovieResponse(initialFetch);
    this.props.initialFetchData(movies);
  }

  render() {
    return (
      <div className="App">
       App
      </div>
    );
  }
}


export const mapStateToProps = state => ({
  initialMovies: state
});

export const mapDispatchToProps = dispatch => ({
  initialFetchData: movies => dispatch(fetchInitialMovies(movies))
});

export default connect(mapStateToProps, mapDispatchToProps)(App);

App.propTypes = {
  initialFetchData: PropTypes.func
};
