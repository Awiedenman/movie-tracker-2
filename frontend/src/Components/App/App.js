import React from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';
import Header from '../../Containers/Header/Header';
import Home from '../../Containers/Home/Home';
import Login from '../../Containers/Login/Login';
import SignUp from '../../Containers/SignUp/SignUp';
import Favorites from '../../Containers/Favorites/Favorites';

import './App.css';

export const App = () => {
  return (
    <main className="grid-container">
      <Header />
      <section className="main-container">
        <Switch>
          <Route exact path="/" component={Home} />
          <Route exact path="/login" component={Login} />
          <Route exact path="/sign-up" component={SignUp} />
          <Route exact path="/favorites" component={Favorites} />
          <Redirect to="/" component={Home}/>
        </Switch>
      </section>
    </main>
  );
};

export default App;