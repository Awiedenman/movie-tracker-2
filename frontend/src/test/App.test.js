import React from 'react';
import { shallow } from 'enzyme';
import { App, mapDispatchToProps, mapStateToProps } from '../Containers/App/App';
import { fetchInitialMovies } from '../Actions';
import { mockInitialMovieResponse } from '../mock-data/mock-responses';
import { mockCleanedMovieList } from '../mock-data/mock-clean-data';

describe('App', () => {
  let wrapper;
  const mockFunc = jest.fn();
  beforeEach(() => wrapper = shallow(<App initialFetchData={mockFunc}/>));


  describe('App', () => {
    test('should ', async () => {
      await wrapper.instance().componentDidMount();
      expect(mockFunc).toHaveBeenCalled();
    });

    test('renders without crashing', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });


  describe('mapDispatchToProps', () => {
    test('should call dispatch when MDTP is called', () => {
      const mockDispatch = jest.fn();
      const actionToDipatch = fetchInitialMovies(mockInitialMovieResponse);
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.initialFetchData(mockInitialMovieResponse);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDipatch);

    });
  });

  describe('mapStateToProps', () => {
    test('should return a props object with a movies array', () => {
      const expected = {
        movies: mockCleanedMovieList
      };

      const mockMappedState = mapStateToProps(mockCleanedMovieList);

      expect(mockMappedState).toEqual(expected);
    });
  });
});


