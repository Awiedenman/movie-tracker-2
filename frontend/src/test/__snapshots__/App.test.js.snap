// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter(Connect(Component)) />,
        <section
          className="main-container"
>
          <Switch>
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/"
                    />
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/login"
                    />
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/sign-up"
                    />
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/favorites"
                    />
                    <Redirect
                              component={[Function]}
                              push={false}
                              to="/"
                    />
          </Switch>
</section>,
      ],
      "className": "grid-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Switch>
            <Route
                        component={[Function]}
                        exact={true}
                        path="/"
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/login"
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/sign-up"
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/favorites"
            />
            <Redirect
                        component={[Function]}
                        push={false}
                        to="/"
            />
</Switch>,
          "className": "main-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/login"
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/sign-up"
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/favorites"
/>,
              <Redirect
                component={[Function]}
                push={false}
                to="/"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/login",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/sign-up",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/favorites",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "push": false,
                "to": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "section",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(Connect(Component)) />,
          <section
            className="main-container"
>
            <Switch>
                        <Route
                                    component={[Function]}
                                    exact={true}
                                    path="/"
                        />
                        <Route
                                    component={[Function]}
                                    exact={true}
                                    path="/login"
                        />
                        <Route
                                    component={[Function]}
                                    exact={true}
                                    path="/sign-up"
                        />
                        <Route
                                    component={[Function]}
                                    exact={true}
                                    path="/favorites"
                        />
                        <Redirect
                                    component={[Function]}
                                    push={false}
                                    to="/"
                        />
            </Switch>
</section>,
        ],
        "className": "grid-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Switch>
              <Route
                            component={[Function]}
                            exact={true}
                            path="/"
              />
              <Route
                            component={[Function]}
                            exact={true}
                            path="/login"
              />
              <Route
                            component={[Function]}
                            exact={true}
                            path="/sign-up"
              />
              <Route
                            component={[Function]}
                            exact={true}
                            path="/favorites"
              />
              <Redirect
                            component={[Function]}
                            push={false}
                            to="/"
              />
</Switch>,
            "className": "main-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
/>,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/login"
/>,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/sign-up"
/>,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/favorites"
/>,
                <Redirect
                  component={[Function]}
                  push={false}
                  to="/"
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/login",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/sign-up",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/favorites",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "push": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "section",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
